libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/exception/ClientException.kt

```diff
@@ -12,7 +12,11 @@ import org.matrix.rustcomponents.sdk.ClientException as RustClientException
 
 fun Throwable.mapClientException(): ClientException {
     return when (this) {
-        is RustClientException.Generic -> ClientException.Generic(msg)
+        is RustClientException -> {
+            when (this) {
+                is RustClientException.Generic -> ClientException.Generic(msg)
```

Comment:
Maybe use `is RustClientException.Generic` directly in the first branch?

Reply:
This is not equivalent, using a dedicated `when` ensures that all the `RustClientException` sub-classes are mapped.